lib/transaction.js

```diff
@@ -234,6 +235,23 @@ class Transaction extends EventEmitter {
         }
       });
   }
+
+  then(onResolve, onReject) {
+    return this._promise.then(onResolve, onReject);
+  }
+
+  catch(onReject) {
+    return this._promise.catch(onReject);
+  }
+
+  finally(onFinally) {
+    return this._promise.finally(onFinally);
+  }
```

Comment:
Might want to do this instead:

```
async finally(onFinally) {
  try {
    return await this._promise
  } finally {
    await onFinally();
  }
}
```

This way, we can provide a `.finally(..)` method without having to rely upon the underlying Promise implementation.

(Note: we should probably also verify that `onFinally` is actually a function before invoking it.)

Reply:
Providing working finally when native promises not supports it can confuse user
Also this can introduce unexpected errors:
selectFromDb()
+    .then(x => map(x))
    .finally(() => {})

start throwing

Reply:
Very true.  In that case, we might want to inject this function into `Transaction.prototype` conditionally.  Otherwise, end-users might get confused when their `Transaction` instance _claims_ to have a `.finally(..)` method, but the method always throws an exception.

(Unfortunately, this might also complicate things on the Typescript side...)

Reply:
Typescript typings not affected, because querybuilder extends part of globally defined Promise interface, so finally will be available only if finally enabled

Reply:
>Otherwise, end-users might get confused when their Transaction instance claims to have a
I think user won't do ('finally' in trx), but ok, I will fix that case

Reply:
Yeah -- just saw the Typescript part of the code.  I think you're correct, but I'm gonna triple-check it via a Typescript compiler just to be sure.

But, my other concern still stands: anybody using reflection to see the list of methods will see `.finally(..)` listed.  Example from my `node 8` console:

```
> var Transaction = require('./lib/transaction')
undefined
> "finally" in Transaction.prototype
true
```

Reply:
Sounz good.  On the plus side: we can remove the conditional logic once `node 8` support is dropped 😅

Reply:
done

Reply:
Looks good -- thx!